ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/HardFork/History/Qry.hs

```diff
@@ -337,6 +340,12 @@ evalExprInEra EraSummary{..} = \(ClosedExpr e) -> go e
         guardEnd $ \end -> e < boundEpoch end
         return eraEpochSize
 
+    go (EGenesisWindow expr) = do
+        s <- go expr
+        guard    $ s >= boundSlot eraStart
```

Comment:
do these just match the slot to the proper era or is it possible that we get `Nothing`?

Reply:
This function, `evalExprInEra` runs the given `Expr` in the context of one specific `EraSummary`. In `runQuery`, we then run it against all eras we know something about (see `Summary`, which contains an `EraSummary` for a (non-empty) prefix of all eras), and returns the first one where `evalExprInEra` succeeded:
https://github.com/IntersectMBO/ouroboros-consensus/blob/ae6b5a37a334a6b1e9c02452de5aa4899f00256e/ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/HardFork/History/Qry.hs#L394-L396

Reply:
thanks